wordpress-mobile/AztecEditor-Android
# Issue/610 api26 style lost prepend new line

### Fix #610
A continuation of https://github.com/wordpress-mobile/AztecEditor-Android/pull/606, This implements the right sequences as observed and described in API26+ in #610, so we can re-work the original intent from user when these situations occur:
- prepending a new line to an inline-styled (for example, bold, italic, etc) piece of text should now not make the style be lost (implemented in `API26PrependNewLineOnStyledTextEvent`)
- prepending a new line to a Block-styled (for example, a Header) piece of text should now not make the style be lost (implemented in `API26PrependNewLineOnStyledSpecialTextEvent`)

Also we fixed a bad behavior that would make the `API26InWordSpaceInsertionEvent` sequence be detected when a space is inserted within a Block, but Blocks behave well in this case, so now there are 3 values that can be returned by `isUserOperationObservedInSequence`, as described here:
```
    enum class ObservedOperationResultType {
        SEQUENCE_FOUND,
        SEQUENCE_NOT_FOUND,
        SEQUENCE_FOUND_CLEAR_QUEUE
    }
```
This way, the third enum value `SEQUENCE_FOUND_CLEAR_QUEUE`  indicates the ObservationQueue that even when the sequence has been found, it can clear itself as no action will be taken.

### Test
1. tap bold
2. enter "a"
3. place the cursor to the beginning (left of "a")
4. tap ENTER
5. observe the style is not lost anymore.

Also repeat with H1 instead of Bold:
1. tap H1
2. enter "a"
3. place the cursor to the beginning (left of "a")
4. tap ENTER
5. observe the style is not lost anymore.

Note: if you're on a stock emulator, please use the soft keyboard for this

### Review
@0nko @daniloercoli
